### PR TITLE
Expose registerFilterType

### PR DESCRIPTION
**Pull request by labkey-alan**
I renamed `generateFilterType` to `registerFilterType` and exposed it so it can be imported by consuming applications. I have tested this in the Biologics and assayreport modules and it works.

### CODE REVIEW

File: src/labkey/filter/Types.ts [hunk 1]
```diff
@@ -154,12 +154,23 @@ export const TYPES_BY_JSON_TYPE_DEFAULT: {
...
+ * Creates a FilterType object and stores it in the global URL Map used by Filter.getFilterTypeForURLSuffix
+ * @param displayText The text to display in a filter menu
+ * @param displaySymbol The symbol to display in a filter menu
+ * @param urlSuffix The suffix used when adding the filter to a URL
+ * @param dataValueRequired Boolean used to indicate if a data value is required for the filter type
+ * @param multiValueSeparator The separator to use if multiple values are allowed for the filter type
+ * @param longDisplayText The text to display in a filter help menu
+ * @param minOccurs The minimum number of times the filter can be applied
+ * @param maxOccurs The maximum number of times the filter can be applied
+ * @param tableWise
```
Comment (labkey-susanh) [inline src/labkey/filter/Types.ts]: Any idea what this parameter means?

Comment (labkey-alan) [reply src/labkey/filter/Types.ts]: Nope, that is why I left it blank. If anyone knows I'll add the appropriate documentation to the parameter.

Comment (labkey-alan) [reply src/labkey/filter/Types.ts]: I added a TODO to document the param, I am going to merge and release. If we figure out what it means we can open a new PR.

**Comment by labkey-kevink** [comment]
Tablewise means the filter is not associated with a single column but
instead applies to all columns on the table.  So far, the `~q` filter is
the only tablewise filter.

Kevin

**Pull request merged**